Dehistory: Panic when fetching dehistory history segment

Started a new network using vegacapsule, performed a few transactions , created a new datanode and ran the below command to fetch a history segment and received a panic message.

**Note:** The history segment I am trying to fetch is a valid segment.

**Error:**

```
./vega datanode dehistory fetch QmSx3VrqEYGu1vYp9iL4fERW8vFWihznSuasP4B9NEMK4t 20 --home /Users/muthupandianvadivelu/muthu/vega/repos/networkdata/testnet/data/node3/
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x101652f10]

goroutine 1 [running]:
code.vegaprotocol.io/vega/datanode/sqlstore.ConnectionConfig.GetPoolConfig({{0x0, 0x0}, 0x0, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...})
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/datanode/sqlstore/config.go:94 +0xc0
code.vegaprotocol.io/vega/datanode/sqlstore.NewTransactionalConnectionSource(0xc7?, {{0x0, 0x0}, 0x0, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...})
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/datanode/sqlstore/connection_source.go:62 +0x98
code.vegaprotocol.io/vega/cmd/data-node/commands/dehistory.verifyChainID(0x14001408c00, {{0x0, 0x0}, 0x0, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...)
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/cmd/data-node/commands/dehistory/fetch.go:93 +0xa4
code.vegaprotocol.io/vega/cmd/data-node/commands/dehistory.(*fetchCmd).Execute(0x105552938, {0x14001fa60c0, 0x2, 0x4?})
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/cmd/data-node/commands/dehistory/fetch.go:50 +0x45c
github.com/jessevdk/go-flags.(*Parser).ParseArgs(0x14001fa6060, {0x140001a8020, 0x6, 0x6})
	/Users/muthupandianvadivelu/go/pkg/mod/github.com/jessevdk/go-flags@v1.4.0/parser.go:316 +0x680
github.com/jessevdk/go-flags.(*Parser).Parse(...)
	/Users/muthupandianvadivelu/go/pkg/mod/github.com/jessevdk/go-flags@v1.4.0/parser.go:186
code.vegaprotocol.io/vega/cmd/data-node/commands.Execute({0x103815168, 0x1400018a000})
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/cmd/data-node/commands/root.go:56 +0x1ac
main.(*datanodeCmd).Execute(...)
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/cmd/vega/datanode.go:27
main.Main({0x103815168, 0x1400018a000})
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/cmd/vega/main.go:59 +0x1d8
main.main()
	/Users/muthupandianvadivelu/muthu/vega/repos/vega/cmd/vega/main.go:45 +0x30
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,7 +14,7 @@
 - [](https://github.com/vegaprotocol/vega/issues/xxxx) -
 
 ### 🐛 Fixes
-- [](https://github.com/vegaprotocol/vega/issues/xxxx) -
+- [7207](https://github.com/vegaprotocol/vega/issues/7207) - Fix panic, return on error in pool configuration
 
 ##  0.65.1
 

--- a/datanode/sqlstore/config.go
+++ b/datanode/sqlstore/config.go
@@ -91,11 +91,12 @@ func (conf ConnectionConfig) GetConnectionStringForPostgresDatabase() string {
 
 func (conf ConnectionConfig) GetPoolConfig() (*pgxpool.Config, error) {
 	cfg, err := pgxpool.ParseConfig(conf.GetConnectionString())
-	cfg.MaxConnLifetime = conf.MaxConnLifetime.Duration
-	cfg.MaxConnLifetimeJitter = conf.MaxConnLifetimeJitter.Duration
 	if err != nil {
 		return nil, err
 	}
+	cfg.MaxConnLifetime = conf.MaxConnLifetime.Duration
+	cfg.MaxConnLifetimeJitter = conf.MaxConnLifetimeJitter.Duration
+
 	cfg.ConnConfig.RuntimeParams["application_name"] = "Vega Data Node"
 	return cfg, nil
 }

